src/components/PublishArticle.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import Preview from "./Preview";
+import { Container, Grid, Form, Dropdown, Button } from "semantic-ui-react";
+
+const PublishArticle = (props) => {
+  const [selectedArticle, setSelectedArticle] = useState();
+  const [message, setMessage] = useState("");
+  const [radio, setRadio] = useState("free");
+  let headers = JSON.parse(localStorage.getItem("J-tockAuth-Storage"));
+  headers = {
+    ...headers,
+    "Content-type": "application/json",
+    Accept: "application/json",
+  };
+
+  const cats = [
+    "Other",
+    "Sport",
+    "Local",
+    "Politics",
+    "Economy",
+    "World",
+    "Entertainment",
+  ];
+
+  const categories = cats.map((cat) => {
+    return { key: cat, text: cat, value: cat.toLowerCase() };
+  });
+
+  useEffect(() => {
+    const fetchSelectedArticle = async () => {
+      try {
+        const response = await axios.get(
+          `/admin/articles/${props.match.params.id}`,
+          {
+            headers: headers,
+          }
+        );
+        setSelectedArticle(response.data.article);
+      } catch (error) {
+        setMessage(error.response.data.message);
+      }
+    };
+    fetchSelectedArticle();
+  }, []);
+
+  const onSubmitHandler = async () => {
+    try {
+      const response = await axios.put(
+        `/admin/articles/${props.match.params.id}`,
+        {
+          headers: headers,
+          params: {
+            activity: "PUBLISH",
+            premium: radio === "premium" ? true : false,
+            category: document
+              .getElementById("category")
+              .firstElementChild.innerText.toLowerCase(),
+          },
+        }
```

Comment:
This is not working, the put arguments should be axios.put("url",{ ...params },{ headers: headers }).

Reply:
Fixed

Reply:
The params should not be wrapped in params: however.

Reply:
I would extract this functionality as well. You have the `fetchWrapper` (not the best name in my opinion, but that is besides that point) This should also be kept outside of this component and reused if necessary.

Reply:
Exactly which functionality ? onSubmitHandler() ?  or creation of params and headers ?
this Submithandler is a put request and it is the only one we have